test_unit.py

```diff
@@ -1,4144 +1,31 @@
-                b"\xbf\xed\xf2\xc1",
-                [
-                    "account",
-                    "pay",
-                    "asset",
-                    "application",
-                    ABIType.from_string("bool"),
-                ],
-                "void",
-                2,
-            ),
-        ]
-
-        for test_case in test_cases:
-            m = Method.from_signature(test_case[0])
-
-            # Check method signature
-            self.assertEqual(m.get_signature(), test_case[0])
-            # Check selector
-            self.assertEqual(m.get_selector(), test_case[1])
-            # Check args
-            self.assertEqual([(a.type) for a in m.args], test_case[2])
-            # Check return
-            self.assertEqual(m.returns.type, test_case[3])
-            # Check txn calls
-            self.assertEqual(m.get_txn_calls(), test_case[4])
-
-    def test_interface(self):
-        test_json = '{"name": "Calculator","desc":"This is an example interface","methods": [{ "name": "add", "returns": {"type": "void"}, "args": [ { "name": "a", "type": "uint64", "desc": "..." },{ "name": "b", "type": "uint64", "desc": "..." } ] },{ "name": "multiply", "returns": {"type": "void"}, "args": [ { "name": "a", "type": "uint64", "desc": "..." },{ "name": "b", "type": "uint64", "desc": "..." } ] }]}'
-        i = Interface.from_json(test_json)
-        self.assertEqual(i.name, "Calculator")
-        self.assertEqual(i.desc, "This is an example interface")
-        self.assertEqual(
-            [m.get_signature() for m in i.methods],
-            ["add(uint64,uint64)void", "multiply(uint64,uint64)void"],
-        )
-
-    def test_contract(self):
-        test_json = '{"name": "Calculator","desc":"This is an example contract","networks":{"wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=":{"appID":1234},"SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=":{"appID":5678}}, "methods": [{ "name": "add", "returns": {"type": "void"}, "args": [ { "name": "a", "type": "uint64", "desc": "..." },{ "name": "b", "type": "uint64", "desc": "..." } ] },{ "name": "multiply", "returns": {"type": "void"}, "args": [ { "name": "a", "type": "uint64", "desc": "..." },{ "name": "b", "type": "uint64", "desc": "..." } ] }]}'
-        c = Contract.from_json(test_json)
-        self.assertEqual(c.name, "Calculator")
-        self.assertEqual(c.desc, "This is an example contract")
-        self.assertEqual(
-            c.networks,
-            {
-                "wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=": NetworkInfo(
-                    app_id=1234
-                ),
-                "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=": NetworkInfo(
-                    app_id=5678
-                ),
-            },
-        )
-        self.assertEqual(
-            [m.get_signature() for m in c.methods],
-            ["add(uint64,uint64)void", "multiply(uint64,uint64)void"],
-        )
-
 
 if __name__ == "__main__":
```

Comment:
For those with more Python experience:  If we move away from requiring a _driver_ to run the tests, is it more idiomatic to co-locate non-cucumber unit test files with source files (named as `<test_name>_test.py` rather than the separate dir?

From my scan of https://docs.python.org/3/library/unittest.html#test-discovery, it seems `_test.py` is an accepted convention.

Reply:
I personally prefer a separate `tests` directory, but we seem to be mostly favoring the approach you mentioned. `pytest` is smart enough to figure it out if you give it the root directory.

Reply:
That being said, I must confess again to not actually trying it... I commit to trying this by standup if no one else has.

Reply:
So is everyone in favor of removing the driver file and using something like: `pytest tests/unit_tests`?

I can also change the test file names to `*_tests.py` to follow the convention.

Reply:
> So is everyone in favor of removing the driver file and using something like: pytest tests/unit_tests?

On my side, yes.  It feels less error prone because tests won't require registration.

> I can also change the test file names to *_tests.py to follow the convention.

* I don't feel strongly here.  I also don't know what's considered common convention.  I originally raised it when considering co-location with source files.
* If it's considered _Pythonic_ and you're up for the change, then I'm all for it.

Reply:
(voting for removal)

Reply:
> So is everyone in favor of removing the driver file and using something like: `pytest tests/unit_tests`?

+1 to getting rid of `test_unit.py`. I think because our unit tests are written using `unittest`, not `pytest`, you may need to use an approach like this: https://stackoverflow.com/a/15630454, but regardless, auto discovery of all tests should be preferred over a driver file.
 
> I can also change the test file names to `*_tests.py` to follow the convention.

I think it only makes sense to add a suffix like this when tests are co-located with other source files.

Reply:
Sounds good, it looks like both of these commands will work:
```
$ python -m unittest discover tests/unit_tests
```
with pytest (better formatting): 
```
$ pytest tests/unit_tests
```

Pytest seems to work with the version that's currently in the `requirements.txt` and outputs better formatting so I'd be inclined to use that.

Reply:
Ok great, I agree `pytest` should be preferred